web-api-shared/src/elastic.rs

```diff
@@ -419,30 +420,38 @@ impl Client {
             .await
     }
 
-    pub async fn search_request<I>(&self, mut body: JsonObject) -> Result<ScoreMap<I>, Error>
+    pub async fn search_request<I, E, F>(
+        &self,
+        mut body: JsonObject,
+        parse_id: F,
```

Comment:
We can't use `DeserializeOwned` here as we have a ES specific string representation for `SnippetId` with `snippet_idx > 0` which we don't really want to use in other interface places (as it might require the customer to parse/split ids and some other reasons).

So instead we allow passing a custom parse function here.